Replace () with Msg containing NoOp value

xypnox opened the pull request.

xypnox committed:
Replace () with Msg containing NoOp value

amitu commented on frontend/Pages/Index.elm:
```diff
@@ -11,7 +11,7 @@ main =
 
 
-app : R.App Config Config ()
+app : R.App Config Config Msg
```
Lets also use a `Model` here, instead of reusing `Config` as `Model`.

xypnox committed:
Use Model instead of Config for types

amitu commented on frontend/Pages/Index.elm:
```diff
@@ -11,9 +11,14 @@ main =
+app : R.App Config Model Msg
 app =
-    R.App config R.init0 R.update0 R.sub0 document
+    R.App config init R.update0 R.sub0 document
```
Lets remove ` R.update0 R.sub0` too.

xypnox replied:
I have added the update and sub too. I have also switched their type definitions from type variable to the Model type used in the file.

xypnox committed:
Add init update sub, resturcture the code

init, update and sub have been added with minimal code.

The code has been restructred to move the main at the end and in a way
that the user types it from top to bottom.

amitu commented on frontend/Pages/Index.elm:
```diff
@@ -35,6 +21,21 @@ type Msg
+
+update : R.In -> Msg -> Model -> ( Model, Cmd (R.Msg Msg) )
+update _ _ model =
+    ( model, Cmd.none )
```
Rename the variable `model` to `m`.

xypnox committed:
Rename model to m

xypnox committed:
Move app to the top
